Apps/Src/UpdateLights.c

```diff
@@ -0,0 +1,62 @@
+#include "Lights.h"
+#include "Tasks.h"
+#define BRAKETHRESHOLD 3
+
+/**
+ * @brief This task waits on the Update Lights Semaphore, then reads the switches that pertain to the lights from the global carstate
+ * (UpdateSwitches will have already updated this). Using those values it uses the lights driver to set the lights to the pertinent values.
+ * 
+ * @param p_arg 
+ */
+void Task_UpdateLights(void* p_arg){
+    car_state_t* GlobalCarState = (car_state_t*) p_arg;
+    switch_states_t* GlobalSwitchStates = &(GlobalCarState->SwitchStates);
+    //Note: the below code is based off the switch_states_t struct defined in the RTOS_ReadSwitches branch rather than main
+    light_switches_t* GlobalLightSwitches = &(GlobalSwitchStates->lightSwitches);
+    OS_ERR err;
+    CPU_TS ts;
+    while(1){
+        OSSemPend(
+            &LightsChange_Sem4,
+            0,
+            OS_OPT_PEND_BLOCKING,
+            &ts,
+            &err
+        );
+        if(GlobalLightSwitches->HEADLIGHT_SW == ON){
+            Lights_Set(Headlight_ON,ON);
+        } else {
+            Lights_Set(Headlight_ON,OFF);
+        }
+
+        if(GlobalLightSwitches->HZD_SW == ON || GlobalLightSwitches->LEFT_SW == ON || GlobalLightSwitches->RIGHT_SW ==ON){
+            OSSemPost(
+                &BlinkLight_Sem4,
+                OS_OPT_POST_ALL,
+                &err
```

Comment:
Where is this semaphore initialized? 

Also, I don't think this is the best way to do this. Why not just have the blink lights task run constantly at 1.5 Hz and read `GlobalLightSwitches` directly?